suggested-wiki-changes/Build-OpenBoard-on-Ubuntu-with-Qt6.md

```diff
@@ -0,0 +1,101 @@
+
+Presumably a new version of the QtWebEngine build document will be required for users
+who require a build under with proprietary codec support under Qt6, this new document 
+version does not yet exist. 
+
+## Prepare your environment
+
+### Install Qt6, qmake6 and OpenBoard dependencies
+
+OpenBoard's instructions for building against Qt5.15.X provide separate sections for
+installation of Qt, qmake and the dependencies of the OpenBoard software itself.
+
+For simplicity, all installations are bundled in a single command here:
+
+```
+sudo apt install build-essential libgl1-mesa-dev libssl-dev libpoppler-dev libpoppler-cpp-dev libpoppler-private-dev libavcodec-dev libavformat-dev libswscale-dev libpaper-dev libva-dev libxcb-shm0-dev libasound2-dev libx264-dev libvpx-dev libvorbis-dev  libtheora-dev libmp3lame-dev libsdl1.2-dev libopus-dev  libfdk-aac-dev libass-dev liblzma-dev libbz2-dev libquazip5-dev libxcb-shape0-dev libxcb-xfixes0-dev qt6-tools-dev qt6-svg-dev qt6-multimedia-dev qt6-webengine-dev libquazip1-qt6-dev
+```
+
+After this installation has completed, run the following command to verify the version number and home directory for the Ubuntu Qt build:
+
+```
+qmake6 -v
+```
+
+on the system(s) where this process has been tested to date this command reports something like:
+
+```
+QMake version 3.1
+Using Qt version 6.4.2 in /usr/lib/x86_64-linux-gnu
+```
+
+## Download OpenBoard sources
+```
+# place yourself in the directory you want 
+# (in this example at user's home directory)
+cd
+
+# create the folder
+mkdir openboard
+
+# enter in it
+cd openboard
+
+# download openboard sources
+https://github.com/tim-littlefair/OpenBoard.git
+```
+
+**Important note** : A pull request will be submitted to the upstream OpenBoard project
+containing this document and the minor changes to OpenBoard.pro required to get the build 
+working - if/when the upstream maintainers accept and merge this PR the sources should
+be downloaded using 
+```
+git clone https://github.com/OpenBoard-org/OpenBoard.git
+```
+instead
+
+## Compile OpenBoard
+
+
+
+Once the installations and git clone operation are complete, you should be
+able to continue by running the following commands:
+```
+qmake6 OpenBoard.pro
+make
+``` 
+
+and then continuing with
+[the compile instructions in OpenBoard's original document](Build-OpenBoard-on-Ubuntu.md#compile-openboard).
+
+As of the current version the following error messages are returned from the 'qmake6' command:
+> Info: creating stash file /home/tim/github/OpenBoard/.qmake.stash
+> Cannot read /home/tim/github/OpenBoard/version.txt: No such file or directory
+> /bin/sh: 1: Syntax error: "&" unexpected
+> /bin/sh: 1: Syntax error: "&" unexpected
+> /bin/sh: 1: Syntax error: "&" unexpected
+These do not appear to reflect a fatal error - qmake6 generates a 
+Makefile and the 'make' command succeeds.
```

Comment:
You based your changes on the `master` branch but included a change to the project file which is coming with 1.7.1. I think you should currently base it on `dev`, which includes the `version.txt` file, and also direct your PR to the `dev` branch.

In between the version info was extracted from the main project file to a new file `version.txt` and an additional CMake based build system was added. The new `version.txt` allows that the version info is used by both build systems.

The error above is non-fatal, but the correct version number will then not be displayed in the "Preferences" and the build-in update check will most probably fail.

Reply:
Thanks for the advice - I've updated my fork to base from the upstream 'dev' branch and to remove the libquazip5-dev package from the dependency list in the Qt document, but apologies if I haven't reworked the repo exactly as you need - I'm happy to withdraw the PR and start from scratch if you want the changes but this PR isn't exactly right for merging them.

The three 'Syntax error: "&&" unexpected' are still appearing for the qmake6 command but the version.txt complaint has gone away.

Reply:
See https://github.com/OpenBoard-org/OpenBoard/pull/627 for the thre `& errors. This is known, the fix is available, but still not applied. But you <fixed what I was proposing.

Reply:
I've verified that after installing from the .deb the version string displayed on the preferences page agrees with the attributes in version.txt.

Are there any tests you'd like me to do (presumably with changes to version.txt) to verify the update behaviour?